### ISSUE TITLE
The code may have some bugs

### ISSUE THREAD

**Issue by li624120638** [open-mmlab/mmdetection#10722]
mmdet/models/data_preprocessors/data_preprocessor.py
DetDataPreprocessor._get_pad_shape():
`
        elif isinstance(_batch_inputs, torch.Tensor):
            ...
            pad_h = int(
                np.ceil(_batch_inputs.shape[1] / self.pad_size_divisor)) * self.pad_size_divisor
            pad_w = int( np.ceil(_batch_inputs.shape[2] / self.pad_size_divisor)) * self.pad_size_divisor
           ...
`
might be
`
elif isinstance(_batch_inputs, torch.Tensor):
            ...
            pad_h = int(
                np.ceil(_batch_inputs.shape[2] / self.pad_size_divisor)) * self.pad_size_divisor
            pad_w = int( np.ceil(_batch_inputs.shape[3] / self.pad_size_divisor)) * self.pad_size_divisor
           ...
`

**Comment by hhaAndroid** [comment]
@li624120638  Yes, thank you very much for your feedback. May I ask if you can submit a pull request to fix it?
